include/aws/http/private/h2_connection.h

```diff
@@ -141,4 +141,9 @@ int aws_h2_connection_on_stream_closed(
     enum aws_h2_stream_closed_when closed_when,
     int aws_error_code);
 
+/* Transform the request to h2 style headers */
+struct aws_http_headers *aws_h2_create_headers_from_request(
+    struct aws_http_message *request,
```

Comment:
does request already have an allocator object on it?

Reply:
Yeah, but aws_http_message is not visible from here, we need another AWS_HTTP_API to get the allocator from it. Do I just add one to get the allocator?

Reply:
I thought about it before, but I am not sure about the API change. So I just let the caller give me an allocator, instead of taking the allocator from request.

Reply:
this seems fine